src/fpm_targets.f90

```diff
@@ -268,7 +268,15 @@ subroutine build_target_list(targets,model)
 
                 case (FPM_UNIT_PROGRAM)
 
-                    call add_target(targets,package=model%packages(j)%name,type = FPM_TARGET_OBJECT,&
+                    if (str_ends_with(lower(sources(i)%file_name), [".c"])) then
+                        exe_type = FPM_TARGET_C_OBJECT
+                    else if (str_ends_with(lower(sources(i)%file_name), [".cpp", ".cc "])) then
+                        exe_type = FPM_TARGET_CPP_OBJECT
+                    else    ! Default to a Fortran object
+                        exe_type = FPM_TARGET_OBJECT
+                    end if
+
+                    call add_target(targets,package=model%packages(j)%name,type = exe_type,&
                                 output_name = get_object_name(sources(i)), &
                                 source = sources(i) &
```

Comment:
Not 100% sure, but I think #777 might be because of a missing macros line here:

```suggestion
                                source = sources(i), &
                                macros = model%packages(j)%macros &
```

Does that seem right @arteevraina?

Reply:
Yes, adding that line should resolve it for the executables. Currently, macros were not passed to executables.

Reply:
Thanks guys for this. @LKedward can you merge this PR and I'll open a new one addressing #777